multisrc/overrides/mangathemesia/thunderscans/src/ThunderScans.kt

```diff
@@ -0,0 +1,120 @@
+package eu.kanade.tachiyomi.extension.ar.thunderscans
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.widget.Toast
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.extension.BuildConfig
+import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ThunderScans : MangaThemesia(
+    "ThunderScans",
+    "https://thunderscans.com",
+    "ar",
+    dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale("ar")),
+) {
+    private val defaultBaseUrl = "https://thunderscans.com"
+
+    override val baseUrl by lazy { getPrefBaseUrl() }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override val client: OkHttpClient = super.client.newBuilder()
+        .rateLimit(1)
+        .build()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val request = super.searchMangaRequest(page, query, filters)
+        if (query.isBlank()) return request
+
+        val url = request.url.newBuilder()
+            .removePathSegment(0)
+            .removeAllQueryParameters("title")
+            .addQueryParameter("s", query)
+            .build()
+
+        return request.newBuilder()
+            .url(url)
+            .build()
+    }
+
+    override fun searchMangaNextPageSelector() = "a[rel=next]"
+
+    override val seriesArtistSelector = "span:contains(الناشر) i"
+    override val seriesAuthorSelector = "span:contains(المؤلف) i"
+    override val seriesGenreSelector = "span:contains(التصنيف) a, .mgen a"
+    override val seriesTypeSelector = "span:contains(النوع) a"
+    override val seriesStatusSelector = "span:contains(الحالة)"
+
+    override fun pageListParse(document: Document): List<Page> {
+        val scriptContent = document.selectFirst("script:containsData(ts_reader)")!!.data()
+        val jsonString = scriptContent.substringAfter("ts_reader.run(").substringBefore(");")
+        val tsReader = json.decodeFromString<TSReader>(jsonString)
+        val imageUrls = tsReader.sources.firstOrNull()?.images ?: return emptyList()
+        return imageUrls.mapIndexed { index, imageUrl -> Page(index, document.location(), imageUrl) }
+    }
```

Comment:
The `MangaThemesia` base class can already handle pages from `ts_reader`.

Reply:
ah ok thanks